### PR TITLE
[corlib] ThreadAbortException protection for ArraySortHelper

### PR DESCRIPTION
**Pull request by lambdageek**
Bump `external/corert`

The ArraySortHelper catches all exceptions in the comparer.
If the sorting thread is aborted while it is running the comparer, the
ArraySortHelper will catch the TAE and propagate an InvalidOperationException
instead.

As a workaround, catch and rethrow TAEs separately.

Add regression test

Related to mono/mono#15418

### CODE REVIEW

**Review by CoffeeFlux — APPROVED** [review]

**Review by steveisok — APPROVED** [review]

**Comment by lambdageek** [comment]
Manual backport to 2020-02 in https://github.com/mono/mono/pull/20468

File: mono/tests/abort-sort.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+using System;
```
Comment (vargaz) [inline mono/tests/abort-sort.cs:1]: This should be a bcl test i think.

Comment (lambdageek) [reply mono/tests/abort-sort.cs]: Ok, changed it to a BCL test

**Comment by lambdageek** [comment]
wasm build is failing due to
```
CSC     [wasm] wasm_corlib_test.dll
Test/System/ArrayTest.cs(3786,3): error CS0619: 'Thread.Abort()' is obsolete: 'Thread.Abort is not supported on the current platform.'
```

can I just wrap the whole thing in `#if !WASM`?

**Review by vargaz — APPROVED** [review]

**Pull request merged**